src/util/std_code.h

```diff
@@ -631,6 +631,78 @@ inline code_assertt &to_code_assert(codet &code)
   return ret;
 }
 
+/// A `codet` representing the declaration that an input of a particular
+/// description has a value which corresponds to the value of a given expression
+/// (or expressions).
+/// When working with the C front end, calls to the `__CPROVER_input` intrinsic
+/// can be added to the input code in order add instructions of this type to the
+/// goto program.
+/// The first argument is expected to be a C string denoting the input
+/// identifier. The second argument is the expression for the input value.
```

Comment:
In line with the discussion with @kroening above: I think you will need to provide this kind of documentation for each of the constructors. It's really no clear how the above maps to the two, very different constructors.

Reply:
Ah. It wasn't clear to me what change @kroening wanted. I will add doxygen for each of the constructors.

Reply:
I have added doxygen for each of the constructors.